plugin/plugin.go

```diff
@@ -54,9 +62,22 @@ type Registration struct {
 	ID       string
 	Config   interface{}
 	Requires []Type
-	Init     func(*InitContext) (interface{}, error)
 
-	added bool
+	// InitFn is called when initializing a plugin. The registration and
+	// context are passed in. The init function may modify the registration to
+	// add exports, capabilites and platform support declarations.
+	InitFn func(*InitContext) (interface{}, error)
```

Comment:
Why the rename? :trollface:

Reply:
Good question: I wanted to hook into `(*Registration).Init`, so I renamed the field `InitFn`. It would have been easier to do this in a single place, but I wanted to break all the usages to make sure everything was kosher.